website/source/docs/builders/cloudstack.html.md

```diff
@@ -0,0 +1,149 @@
+
+There are many configuration options available for the builder. They are
+segmented below into two categories: required and optional parameters. Within
+each category, the available configuration keys are alphabetized.
+
+In addition to the options listed here, a
+[communicator](/docs/templates/communicator.html) can be configured for this
+builder.
+
+### Required:
+
+-   `api_url` (string) - The CloudStack API endpoint we will connect to.
+
+-   `api_key` (string) - The API key used to sign all API requests.
+
+-   `cidr_list` (array) - List of CIDR's that will have access to the new
+    instance. This is needed in order for any provisioners to be able to
+    connect to the instance. Usually this will be the NAT address of your
+    current location. Only required when `use_local_ip_address` is `false`.
+
+-   `instance_name` (string) - The name of the instance. Defaults to a
+    dynamically generated name (using a time ordered UUID).
+
+-   `network` (string) - The name or ID of the network to connect the instance
+    to.
+
+-   `secret_key` (string) - The secret key used to sign all API requests.
+
+-   `service_offering` (string) - The name or ID of the service offering used
+    for the instance.
+
+-   `soure_iso` (string) - The name or ID of an ISO that will be mounted before
+    booting the instance. This option is mutual exclusive with `source_template`.
+
+-   `source_template` (string) - The name or ID of the template used as base
+    template for the instance. This option is mutual explusive with `source_iso`.
+
+-   `template_name` (string) - The name of the new template. Defaults to a
+    dynamically generated name (using a timestamp).
+
+-   `template_display_text` (string) - The display text of the new template.
+    Defaults to the `template_name`.
+
+-   `template_os` (string) - The name or ID of the template OS for the new
+    template that will be created.
+
+-   `zone` (string) - The name or ID of the zone where the instance will be
+    created.
+
+### Optional:
+
+-   `async_timeout` (int) - The time (in seconds) to wait for async calls to
+    finish. Defaults to 1800 seconds.
+
+-   `disk_offering` (string) - The name or ID of the disk offering used for the
+    instance. This option is required when using `source_iso`.
+
+-   `disk_size` (int) - The size (in GB) of the root disk of the new instance.
+    This option is required when using `source_template`.
+
+-   `http_get_only` (boolean) - Some cloud providers only allow HTTP GET calls to
+    their CloudStack API. If using such a provider, you need to set this to `true`
+    in order for the provider to only make GET calls and no POST calls.
+
+-   `hypervisor` (string) - The target hypervisor for the new template. This
+    option is required when using `source_iso`.
+
+-   `keypair` (string) - The name of the SSH key pair that will be used to
+    access the instance.
+
+-   `project` (string) - The name or ID of the project to deploy the instance to.
```

Comment:
Nope, no default... Working in a project is completely optional. So no project is also fine.